app/src/behaviors/behavior_tri_state.c

```diff
@@ -0,0 +1,301 @@
+    if (ev == NULL) {
+        return ZMK_EV_EVENT_BUBBLE;
+    }
+    for (int i = 0; i < ZMK_BHV_MAX_ACTIVE_TRI_STATES; i++) {
+        struct active_tri_state *tri_state = &active_tri_states[i];
+        if (!tri_state->is_active) {
+            continue;
+        }
+        if (tri_state->position == ev->position) {
+            continue;
+        }
+        if (!is_other_key_ignored(tri_state, ev->position)) {
+            LOG_DBG("Tri-State interrupted, ending at %d %d", tri_state->position, ev->position);
+            tri_state->is_active = false;
+            struct zmk_behavior_binding_event event = {.position = tri_state->position,
+                                                       .timestamp = k_uptime_get()};
+            if (tri_state->is_pressed) {
+                behavior_keymap_binding_released(
+                    (struct zmk_behavior_binding *)&tri_state->config->continue_behavior, event);
+            }
+            trigger_end_behavior(tri_state);
+            return ZMK_EV_EVENT_BUBBLE;
+        }
+        if (ev->state) {
+            stop_timer(tri_state);
+        } else {
+            reset_timer(ev->timestamp, tri_state);
+        }
+    }
+    return ZMK_EV_EVENT_BUBBLE;
+}
+
+static int tri_state_layer_state_changed_listener(const zmk_event_t *eh) {
+    struct zmk_layer_state_changed *ev = as_zmk_layer_state_changed(eh);
+    if (ev == NULL) {
+        return ZMK_EV_EVENT_BUBBLE;
+    }
+    if (!ev->state) {
+        return ZMK_EV_EVENT_BUBBLE;
+    }
+    for (int i = 0; i < ZMK_BHV_MAX_ACTIVE_TRI_STATES; i++) {
+        struct active_tri_state *tri_state = &active_tri_states[i];
+        if (!tri_state->is_active) {
+            continue;
+        }
+        if (!is_layer_ignored(tri_state, ev->layer)) {
+            LOG_DBG("Tri-State layer changed, ending at %d %d", tri_state->position, ev->layer);
+            tri_state->is_active = false;
+            struct zmk_behavior_binding_event event = {.position = tri_state->position,
+                                                       .timestamp = k_uptime_get()};
+            if (tri_state->is_pressed) {
+                behavior_keymap_binding_released(
+                    (struct zmk_behavior_binding *)&tri_state->config->continue_behavior, event);
+            }
+            behavior_keymap_binding_pressed(
+                (struct zmk_behavior_binding *)&tri_state->config->end_behavior, event);
+            behavior_keymap_binding_released(
+                (struct zmk_behavior_binding *)&tri_state->config->end_behavior, event);
+            return ZMK_EV_EVENT_BUBBLE;
+        }
+    }
+    return ZMK_EV_EVENT_BUBBLE;
+}
+
+#define _TRANSFORM_ENTRY(idx, node)                                                                \
+    {                                                                                              \
+        .behavior_dev = DT_LABEL(DT_INST_PHANDLE_BY_IDX(node, bindings, idx)),                     \
```

Comment:
- in #1499, `DT_LABEL` was migrated to `DT_PROP`
- and then in #2028, `DT_PROP` was migrated to `DEVICE_DT_NAME`

```suggestion
        .behavior_dev = DEVICE_DT_NAME(DT_INST_PHANDLE_BY_IDX(node, bindings, idx)), \
```